Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm

```diff
@@ -262,41 +292,113 @@ static bool isCopyDisplayedPixelBufferAvailable()
 
 MediaTime MediaPlayerPrivateWebM::currentMediaTime() const
 {
-    MediaTime synchronizerTime = PAL::toMediaTime(PAL::CMTimebaseGetTime([m_synchronizer timebase]));
+    MediaTime synchronizerTime = clampTimeToLastSeekTime(PAL::toMediaTime(PAL::CMTimebaseGetTime([m_synchronizer timebase])));
     if (synchronizerTime < MediaTime::zeroTime())
         return MediaTime::zeroTime();
-    if (synchronizerTime > durationMediaTime())
-        return durationMediaTime();
```

Comment:
Why is it no longer needed?

Reply:
It was never needed. durationMediaTime is the value of max(duration, currentTime) as set by the observer.

It is a copy of MediaPlayerPrivateMediaSourceAVFObjC code